R/nest.R

```diff
@@ -350,7 +350,9 @@ as_df <- function(x, col) {
     x
   } else if (vec_is(x)) {
     # Preserves vec_size() invariant
-    new_data_frame(set_names(list(x), col))
```

Comment:
Could also use `setNames()` here?

Reply:
Davis reimplemented `set_names()` in C in the last CRAN version so maybe this PR can be closed.

Reply:
Even with the update to `set_names()`, `names<-` is still much faster. In this case, `set_names()` basically does a few checks and then eventually calls `names<-` anyways.

``` r
library(rlang)
library(stats)

x <- list(1)
col <- "stuff"

bench::mark(
  set_names(x, col),
  `names<-`(x, col),
  setNames(x, col),
  iterations = 100000
)
#> # A tibble: 3 x 6
#>   expression             min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>        <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 set_names(x, col)   1.18µs   1.42µs   650390.        0B     26.0
#> 2 `names<-`(x, col)    161ns    185ns  4828749.        0B      0  
#> 3 setNames(x, col)     524ns    631ns  1463210.        0B     29.3
```

The best option might be to just pass names through in `new_data_frame()` since you can do this now. I'll update to this approach

```r
library(vctrs)
library(stats)
library(rlang)

x <- list(1)
col <- "stuff"

bench::mark(
  new_data_frame(x, names = col),
  new_data_frame(set_names(x, col)),
  new_data_frame(setNames(x, col)),
  new_data_frame(`names<-`(x, col)),
  iterations = 200000
)
#> # A tibble: 4 x 6
#>   expression                             min median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                        <bch:tm> <bch:>     <dbl> <bch:byt>    <dbl>
#> 1 new_data_frame(x, names = col)       900ns 1.29µs   757483.    6.39KB    114. 
#> 2 new_data_frame(set_names(x, col))   2.12µs 2.43µs   382186.        0B     22.9
#> 3 new_data_frame(setNames(x, col))    1.42µs 1.67µs   567186.        0B     28.4
#> 4 new_data_frame(`names<-`(x, col))   1.04µs 1.21µs   763106.        0B     26.7
```